iroh/src/protocol.rs

```diff
@@ -201,16 +201,27 @@ impl Router {
         &self.endpoint
     }
 
+    /// Checks if the router is already shutdown.
+    pub fn is_shutdown(&self) -> bool {
+        self.cancel_token.is_cancelled()
+    }
+
     /// Shuts down the accept loop cleanly.
     ///
     /// If some [`ProtocolHandler`] panicked in the accept loop, this will propagate
     /// that panic into the result here.
-    pub async fn shutdown(self) -> Result<()> {
+    ///
+    /// If already shutdown, it just returns `Ok`.
```

Comment:
```suggestion
    /// If already shutdown, it returns `Ok`.
```
Text is almost always better off without words like "just" or "only" etc.

Reply:
This should probably say something about what it is waiting on.  Is it fully shut down when it returns?

Reply:
done